key-manager/internal/models/kserve.go

```diff
@@ -78,3 +79,36 @@ func (m *Manager) ListAvailableModels() ([]ModelInfo, error) {
 
 	return modelList, nil
 }
+
+// ListAvailableLLMs lists all LLMInferenceServices across all namespaces.
+func (m *Manager) ListAvailableLLMs() ([]openai.Model, error) {
+	// Define InferenceService GVR
+	llmGVR := schema.GroupVersionResource{
+		Group:    "serving.kserve.io",
+		Version:  "v1alpha1",
+		Resource: "llminferenceservices",
+	}
+
+	log.Printf("DEBUG: Attempting to list LLMInferenceServices with GVR: %+v", llmGVR)
+
+	// List all InferenceServices across all namespaces
+	list, err := m.kuadrantClient.Resource(llmGVR).List(
+		context.Background(), metav1.ListOptions{})
```

Comment:
I think we should propagate standard `context.Context` from the incoming request. At the HTTP boundary, we can extract `ctx := ginCtx.Request.Context()` and pass that down (so that only the handler layer should know about Gin; services/repos should accept `context.Context`). 

Perhaps out of scope for this PR, but we worth tackling sooner than later.